### PR TITLE
Pull request for the integration of pymmcore-plus

### PR DESCRIPTION
**Pull request by felixwanitschke**

### CODE REVIEW

**Comment by jacopoabramo** [comment]
@felixwanitschke the folder `build` and all its content should be deleted entirely and not committed because it's a local copy and it must not go into the main repository

File: .gitignore [hunk 1]
```diff
@@ -12,7 +12,7 @@ __pycache__/
 # Distribution / packaging
 .Python
 env/
-build/
+#build/
```
[the remark this answers is not in the document]
Comment (felixwanitschke) [reply .gitignore:15]: I have de-commented this after the commit but I had to comment it to commit the deleted build folder :)

File: src/napari_live_recording/ui/__init__.py [hunk 2]
```diff
@@ -67,9 +69,9 @@ def addCameraUI(self, interface: str, name: str, idx: int):
                 widget = ComboBox(parameter.options, name)
                 widget.signals["currentTextChanged"].connect(lambda text, name=name: camera.changeParameter(name, text))
             settingsLayout.addRow(widget.label, widget.widget)
-        deleteButton = QPushButton("Delete camera")
-        deleteButton.clicked.connect(lambda: self.deleteCameraUI(cameraKey))
-        settingsLayout.addRow(deleteButton)
+        self.deleteButton = QPushButton("Delete camera")
+        self.deleteButton.clicked.connect(lambda: self.deleteCameraUI(cameraKey))
+        settingsLayout.addRow(self.deleteButton)
```
Comment (jacopoabramo) [inline src/napari_live_recording/ui/__init__.py:74]: Why did you add the `self` here?

File: src/napari_live_recording/ui/__init__.py [hunk 3]
```diff
@@ -83,7 +85,10 @@ def deleteCameraUI(self, cameraKey: str) -> None:
         self.mainController.deleteCamera(cameraKey)
         self.mainLayout.removeRow(self.cameraWidgetGroups[cameraKey])
         del self.cameraWidgetGroups[cameraKey]
-    
+
+    def _enableDeleteButton(self, status: bool):
+        self.deleteButton.setEnabled(not status)
+
```
Comment (jacopoabramo) [inline src/napari_live_recording/ui/__init__.py:91]: Please remove this. I will test on my side and see what happens; with the latest changes I added it should work now

File: src/napari_live_recording/control/devices/pymmcore-plus.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    
+    def changeROI(self, newROI: ROI):
+        if newROI <= self.fullShape:
+            self.roiShape = newROI
+    
+    def close(self) -> None:
+        self.setAcquisitionStatus(False)
+        self.__capture.unloadDevice(self.name)
+        pass
+        #self.__capture.release()
```
Comment (jacopoabramo) [inline src/napari_live_recording/control/devices/pymmcore-plus.py:137]: delete these

**Review by jacopoabramo — CHANGES_REQUESTED** [review]
First review. If possible try also to do a general formatting of the files via an autoformatter like black: https://dev.to/adamlombard/how-to-use-the-black-python-code-formatter-in-vscode-3lo0
